src/ast/nodes/Identifier.ts

```diff
@@ -160,7 +161,11 @@ export default class Identifier extends NodeBase {
 					storeName: true,
 					contentOnly: true
 				});
-				if (this.parent.type === NodeType.Property && (<Property>this.parent).shorthand) {
+				let relevantParent = this.parent;
+				if (relevantParent.type === NodeType.AssignmentPattern) {
+					relevantParent = (<AssignmentPattern>relevantParent).parent;
```

Comment:
Perhaps skip over this path entirely - `if (this.parent.type !== NodeType.AssignmentPatern && ...`. Because surely the parent of an assignment pattern can never be an object shorthand?

Reply:
Quite to the contrary, this is the most important bit! In this example

```js
const {name = 10 } = {};
```

the parent of the assignment pattern *is* a shorthanded property (definitely worth checking this out: hhttp://astexplorer.net/#/gist/a593a01bc9b6b052a1ee898fa4f2c944/91715110ff8c51a2ad9201408ab6f76225b3bc6e )

This was in fact the only situation in which the bug triggered.

Reply:
Not sure how to write this more nicely, though.

Reply:
Ahh I see, looks good then.